masonry/src/passes/update.rs

```diff
@@ -145,3 +145,26 @@ pub(crate) fn run_update_pointer_pass(root: &mut RenderRoot, root_state: &mut Wi
     // Pass root widget state to synthetic state create at beginning of pass
     root_state.merge_up(root.widget_arena.get_state_mut(root.root.id()).item);
 }
+
+// ----------------
+
+pub(crate) fn run_update_scroll_pass(root: &mut RenderRoot) {
+    let _span = info_span!("update_scroll").entered();
+
+    let scroll_request_targets = std::mem::take(&mut root.state.scroll_request_targets);
+    for (target, rect) in scroll_request_targets {
```

Comment:
Does the order matter here? Is it meaningful to document that?

Reply:
The order doesn't matter. `scroll_request_target` was initially an option, and I turned it to a Vec just to cover hypothetical cases where someone would want to pan *several* scroll areas at once.

I guess several targets could interfere if they're in the same scroll region. I think this will be rare enough that we won't need to document it.